tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+
+    api1 = MagicMock(spec=GogoGate2Api)
+    api1.info.return_value = InfoResponse(
+        user="user1",
+        gogogatename="gogogatename0",
+        model="",
+        apiversion="",
+        remoteaccessenabled=False,
+        remoteaccess="321bca.blah.blah",
+        firmwareversion="",
+        apicode="",
+        door1=Door(
+            door_id=1,
+            permission=True,
+            name="Door1",
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.CLOSED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=2,
+            temperature=None,
+        ),
+        door2=Door(
+            door_id=2,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+
+    def new_api(ip_address: str, username: str, password: str) -> GogoGate2Api:
+        if ip_address == "127.0.1.0":
+            return api0
+        if ip_address == "127.0.1.1":
+            return api1
+        raise Exception(f"Untested ip address {ip_address}")
+
+    component_factory.api_class_mock.side_effect = new_api
+
+    await component_factory.configure_component(
+        cover_config=[
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover0",
+                CONF_IP_ADDRESS: "127.0.1.0",
+                CONF_USERNAME: "user0",
+                CONF_PASSWORD: "password0",
+            },
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover1",
+                CONF_IP_ADDRESS: "127.0.1.1",
+                CONF_USERNAME: "user1",
+                CONF_PASSWORD: "password1",
+            },
+        ]
+    )
+    entity_ids = hass.states.async_entity_ids(COVER_DOMAIN)
+    assert entity_ids is not None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)